components/config/src/config/mod.rs

```diff
@@ -395,6 +400,48 @@ impl Default for Config {
     }
 }
 
+/// Used for deserializing values that can be either a single value or a vec of values
+pub(crate) fn might_be_single<'de, T, D>(deserializer: D) -> Result<Vec<T>, D::Error>
+where
+    T: DeserializeOwned,
+    D: Deserializer<'de>,
+{
+    let v = MightBeSingle::deserialize(deserializer)?;
+    Ok(v.into())
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub(crate) enum MightBeSingle<T> {
```

Comment:
If we do support that, we need to add a deprecation message when there is only one value. I'm not entirely we need to though, it's a pretty simple change to do from the user POV.

As for name... `SingleOrVec`?

Reply:
The problem I ran into while testing before adding the backwards-compatibility is that Zola doesn't error out if you leave the old value names in the config file, instead using default values, which is a very bad way to handle it. The user will think they have the configuration option set while it's not until they realize it's broken purely due to a single-letter rename.

I can change the name and add a deprecation message, sure.

Reply:
Update: adding a meaningful deprecation message seems to be much harder than I expected due to the constraints of how serde works